keps/sig-storage/20200212-volume-group.md

```diff
@@ -0,0 +1,823 @@
+  // entries than this number in the response. If the actual number of
+  // entries is more than this number, the Plugin MUST set `next_token`
+  // in the response which can be used to get the next page of entries
+  // in the subsequent `ListGroupSnapshots` call. This field is OPTIONAL. If
+  // not specified (zero value), it means there is no restriction on the
+  // number of entries that can be returned.
+  // The value of this field MUST NOT be negative.
+  int32 max_entries = 1;
+
+  // A token to specify where to start paginating. Set this field to
+  // `next_token` returned by a previous `ListGroupSnapshots` call to get the
+  // next page of entries. This field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string starting_token = 2;
+}
+
+message ListGroupSnapshotsResponse {
+  option (alpha_message) = true;
+
+  message Entry {
+    // This field is REQUIRED
+    GroupSnapshot group_snapshot = 1;
+  }
+
+  repeated Entry entries = 1;
+
+  // This token allows you to get the next page of entries for
+  // `ListGroupSnapshots` request. If the number of entries is larger than
+  // `max_entries`, use the `next_token` as a value for the
+  // `starting_token` field in the next `ListGroupSnapshots` request. This
+  // field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string next_token = 2;
+}
+```
+
+## Proposal for Volume Placement
+
+### API Changes
+
+In order to support Volume Placement, An `AllowedTopologies` field will be added to the VolumeGroupClass API:
+
+```
+type VolumeGroupClass struct {
+        metav1.TypeMeta
+        // +optional
+        metav1.ObjectMeta
+
+        // Driver is the driver expected to handle this VolumeGroupClass.
+        // This value may not be empty.
+        Driver string
+
+        // Parameters holds parameters for driver.
+        // These values are opaque to the system and are passed directly
+        // to the driver.
+        // +optional
+        Parameters map[string]string
+
+        // This field specifies whether group snapshot is supported.
+        // The default is false.
+        // +optional
+        GroupSnapshot *bool
+
+        // Restrict the topologies where a group of volumes can be located.
+        // Each driver defines its own supported topology specifications.
+        // An empty TopologySelectorTerm list means there is no topology restriction.
+        // This field is passed on to the drivers to handle placement of a group of
+        // volumes on storage pools.
+        // +optional
+        AllowedTopologies []api.TopologySelectorTerm
+}
+```
+
+### Example Yaml Files for Volume Placement
+
+A VolumeGroupClass that supports placement:
+```
+apiVersion: volumegroup.storage.k8s.io/v1alpha1
+kind: VolumeGroupClass
+metadata:
+  name: placementGroupClass1
+spec:
+  parameters:
+     …...
+  allowedTopologies: [failure-domain.example.com/placement: storagePool1]
```

Comment:
Is it possible that one VolumeGroupClass has both groupSnapshot and allowedTopologies as parameters?

Reply:
It is possible in the current design.